docs/source/design/ui_design.rst

```diff
@@ -0,0 +1,328 @@
+argument. 
+
+QCEngine
+========
+QCEngine is a general purpose quantum chemistry program interface. It is a
+Python library that provides a common API for quantum chemistry programs.
+QCEngine is an open-source package distributed under the Apache License 2.0.
+
+Below you can find how to run an SCF calculation for a hydrogen molecule using
+QCEngine.
+
+.. code-block:: python
+
+    import qcengine as qcng
+    import qcelemental as qcel
+    mol = qcel.models.Molecule.from_data('H 0. 0. 0. \n H 0. 0. 1.')
+    out = qcng.compute({"molecule": mol, "driver": "energy", "model": {"method":
+    "SCF", "basis": "sto-3g"}}, "NWChemEx")
+    energy = out.return_result
+
+Here, mol is the molecule object (type
+``qcelemental.models.molecule.Molecule``), which is created using the
+``qcel.models.Molecule.from_data()`` function from QCElemental package. The SCF
+energy is computed using the ``qcng.compute()`` function, from QCEngine
+package. Note that, the input for the ``qcng.compute`` function is a Python
+dictionary with a schema defined by QCElemental.
+
+NWChemEx
+========
+
+In NWChemEx, running an SCF calculation for a hydrogen molecule is made easy
+through the use of a Python function with named arguments, as shown below.
+
+.. code-block:: python
+
+    import nwchemex as nwx 
+    energy = nwx.calculate_scf_energy(molecule = 'H 0. 0. 0. \n H 0. 0. 1.', basis = 'sto-3g')
+
+While the example shows only the two required arguments for the ``nwx.calculate_scf_energy()``
+function, the whole function signature is given below: 
+
+.. code-block:: python
+
+    def calculate_scf_energy(molecule: Union[str, chemist.Molecule], basis:
+    Union[str, simde.type.ao_space], spin: int = 1, max_iterations: int = 50,
+    scf_thresh: float = 1e-10, options: Union[Dict, dataclass] = None, **kwargs) -> float:
+
+In this function, the only required arguments are ``molecule``, which can either
+be a Python ``string`` or a ``chemist.Molecule`` object and ``basis``, which can
+either be a a Python ``string`` or a ``simde.type.ao_space``. Additional SCF
+related options are also explicitly defined in the function with reasonable
+default values. While the number of options are limited for SCF, for correlated
+methods more options may need to be defined on top of the SCF related options.
+One solution to avoid a long list of arguments is to provide additional options
+through ``**kwargs`` argument at the end. However, this would require users to
+know the correct keywords. To help our users to figure out the available
+keywords, we also provide the ``options`` argument, which is an object that
+holds all the possible options with reasonable default values. Our users can
+grab this object and figure out the available options and their default values
+and modify them as they wish. 
+
+Similarly, ``calculate_X_energy`` functions will also be provided for all other
+quantum chemistry methods supported by NWChemEx. Users may also request other
```

Comment:
This isn't a static list. Even if it was, it's not well defined, *e.g.*, are MP2 and MP2-F12 different methods, and therefore have different functions? What about MP2 and DLPNO-MP2? What about MP2 and density-fit MP2?...

Also who is responsible for adding these functions?